File: software/examples/javascript/ExampleBeep.js
var Tinkerforge = require('tinkerforge');

var HOST = 'localhost';
var PORT = 4223;
var UID = 'iMv';// Change to your UID

var ipcon = new Tinkerforge.IPConnection();// Create IP connection
var ps = new Tinkerforge.BrickletPiezoSpeaker(UID, ipcon);// Create device object

ipcon.connect(HOST, PORT,
    function(error) {
        console.log('Error: '+error);        
    }
);// Connect to brickd

// Don't use device before ipcon is connected
ipcon.on(Tinkerforge.IPConnection.CALLBACK_CONNECTED,
    function(connectReason) {
        //Make 2 second beep with a frequency of 1kHz
        ps.beep(2000, 1000);      
    }
);

console.log("Press any key to exit ...");
process.stdin.on('data',
    function(data) {
        ipcon.disconnect();
        process.exit(0);
    }
);

File: software/examples/c/example_beep.c
#include <stdio.h>

#include "ip_connection.h"
#include "bricklet_piezo_speaker.h"

#define HOST "localhost"
#define PORT 4223
#define UID "XYZ" // Change to your UID

int main() {
	// Create IP connection
	IPConnection ipcon;
	ipcon_create(&ipcon);

	// Create device object
	PiezoSpeaker ps;
	piezo_speaker_create(&ps, UID, &ipcon); 

	// Connect to brickd
	if(ipcon_connect(&ipcon, HOST, PORT) < 0) {
		fprintf(stderr, "Could not connect\n");
		exit(1);
	}
	// Don't use device before ipcon is connected

	// Make 2 second beep with a frequency of 1kHz
	piezo_speaker_beep(&ps, 2000, 1000);

	printf("Press key to exit\n");
	getchar();
	ipcon_destroy(&ipcon);
}
